src/agent/src/storage/mod.rs

```diff
@@ -380,8 +381,122 @@ fn mount_storage(logger: &Logger, storage: &Storage) -> Result<()> {
         "mount-options" => options.as_str(),
     );
 
+    let confidential = storage.driver_options.contains(&"confidential=true".into());
+    let ephemeral = storage.driver_options.contains(&"ephemeral=true".into());
+
+    let src_path = if confidential && ephemeral {
+        // NOTE: This whole branch will be replaced by the following
+        // upstream script eventually:
+        // https://github.com/confidential-containers/guest-components/blob/main/confidential-data-hub/storage/scripts/luks-encrypt-storage
+
+        // TODO: Ensure that this allows:
+        //  * Multiple containers (same pod) to mount the same ephemeral
+        //    volume.
+        //  * A container to mount the same ephemeral volume mutliple
+        //    times (under different paths).
+        // => No issue in my testing but need to ensure that this is
+        // only called once per pod.
+
+        // Path to the key file that will be passed to the cryptsetup
+        // commands.
+        const KEY_FILE_PATH: &str = "/tmp/ephemeral_encryption_key";
+
+        // Name of the devmapper that will live under /dev/mapper/.
+        let devmapper_device_name = src_path
+            .file_name()
+            .ok_or_else(|| anyhow!("could not get filename from {:?}", src_path))?
```

Comment:
Does this easier to read version work too?

anyhow!("could not get filename from {src_path}")?